src/util/string_constant.cpp

```diff
@@ -32,8 +32,94 @@ string_constantt::string_constantt(
   set("kind", kind);
 }
 
+#if __APPLE__ && __MACH__
+/* Apple enjoys their users suffering to write platform-independent code, that's
+ * why they chose to *not* provide the c(16|32)rtomb functions. */
+
+typedef uint16_t char16_t;
+typedef uint32_t char32_t;
+
```

Comment:
This is causing the error:

```
/Users/user/Repos/esbmc/src/util/string_constant.cpp:39:18: error: cannot combine with previous 'type-name' declaration specifier
typedef uint16_t char16_t;
...
^~~~~~~~~~~~~~~~~~~~~~~~~
/Users/user/Repos/esbmc/src/util/string_constant.cpp:40:18: error: cannot combine with previous 'type-name' declaration specifier
typedef uint32_t char32_t;
```

Removing these lines is enough to make the compilation work.

Reply:
Nice, thanks for testing it Rafael! I've removed those two definitions.